org.lflang/src/lib/core/federated/net_common.h

```diff
@@ -592,6 +600,35 @@ THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
 #define MSG_TYPE_PORT_ABSENT 23
 
 
+
+/**
+ * A message that informs the RTI about connections between this federate and
+ * other federates where messages are routed through the RTI. Currently, this
+ * only includes logical connections when the coordination is centralized. This
+ * information is needed for the RTI to perform the centralized coordination.
+ * 
+ * @note Only information about the immediate neighbors is required. The RTI can
+ * transitively obtain the structure of the federation based on each federate's
+ * immediate neighbor information.
+ *
+ * The next 4 bytes is the number of upstream federates. 
+ * The next 4 bytes is the number of downstream federates.
+ * 
+ * Depending on the first four bytes, the next bytes are pairs of (fed ID (2
+ * bytes), delay (8 bytes)) for this federate's connection to upstream federates
+ * (by direct connection). The delay is the minimum "after" delay of all
+ * connections from the upstream federate.
+ *
+ * Depending on the second four bytes, the next bytes are fed IDs (2
+ * bytes each), of this federate's downstream federates (by direct connection).
+ *
+ * @note The upstream and downstream connections are transmitted on the same
+ *  message to prevent (at least to some degree) the scenario where the RTI has
+ *  information about one, but not the other (which is a critical error).
```

Comment:
It seems that possible critical error is actually a different one: Where federate A claims that federate B is downstream of it, but federate B does not claim that federate A is upstream of it.  Perhaps federates should only inform the RTI of **downstream** federates.  The RTI would use this to populate the data structure for both referenced federates. Then the RTI could never have inconsistent data.

Reply:
I have created an issue to track this problem.